packages/dx-grid-core/src/plugins/integrated-filtering/computeds.js

```diff
@@ -50,27 +58,34 @@ const filterHierarchicalRows = (rows, predicate, getRowLevelKey, isGroupRow) =>
 
 export const filteredRows = (
   rows,
-  filters,
+  filterExpr,
   getCellValue,
   getColumnPredicate,
   isGroupRow,
   getRowLevelKey,
 ) => {
-  if (!filters.length || !rows.length) return rows;
-
-  const predicate = filters.reduce(
-    (prevPredicate, filter) => {
-      const { columnName, ...filterConfig } = filter;
-      const customPredicate = getColumnPredicate && getColumnPredicate(columnName);
-      const columnPredicate = customPredicate || defaultPredicate;
-
-      return (row) => {
-        const result = columnPredicate(getCellValue(row, columnName), filterConfig, row);
-        return result && prevPredicate(row);
-      };
-    },
-    () => true,
-  );
+  if (!filterExpr || !Object.keys(filterExpr).length || !rows.length) return rows;
+
+  const getPredicateFromFilter = (filter) => {
+    const { columnName, ...filterConfig } = filter;
+    const customPredicate = getColumnPredicate && getColumnPredicate(columnName);
+    const columnPredicate = customPredicate || defaultPredicate;
+
+    return row => columnPredicate(getCellValue(row, columnName), filterConfig, row);
+  };
+
+  const predicateGenerator = (filterExpression) => {
```

Comment:
Maybe call this function `getPredicateFromExpression` to keep names consistent? This could also make it more transparent what this thing does.